src/class-wc-gateway-btcpay.php

```diff
@@ -969,21 +970,28 @@ public function ipn_callback()
                     wp_die('Invalid IPN');
                 }
             } catch (\Exception $e) {
-                $error_string = 'IPN Check: Can\'t find invoice ' . $json['id'] . ' (Order ID: ' . $json['orderId'] . ')';
+                $error_string = 'IPN Check: Can\'t find invoice ' . $json['id'] . ' (Order Number: ' . $json['orderId'] . ')';
                 $this->log("    [Error] $error_string");
                 $this->log("    [Error] " . $e->getMessage());
 
                 wp_die($e->getMessage());
             }
 
-            $order_id = $invoice->getOrderId();
+            // The BTCPay order id is the WooCommerce order number, see $invoice->setOrderId in process_payment()
+            $order_number = $invoice->getOrderId();
+
+            // We get the actual WooCommerce ID from the pos data, see $invoice->setPosData in process_payment()
+            // The posData is a string and needs to be JSON decoded separately.
+            $posData = json_decode($json['posData'], true);
+            $order_id = $posData['WooCommerce']['Order ID'];
```

Comment:
Nitpick: I was not sure if spaces in associative arrays work in PHP as I never used it. I always thought spaces are not allowed but turns out this is perfectly fine and supported. I see this is needed because on how the posData is shown, so all good.